### ISSUE TITLE
Problem connecting Flatten() and LSTM()

### ISSUE THREAD

**Issue by Bigpig4396** [keras-team/keras#11506]
my network:

model = Sequential()
model.add(Conv2D(32, (3, 3), input_shape=(map_size, map_size, 1), strides=(1, 1), padding="same"))
model.add(Activation('relu'))
model.add(Flatten())
model.add(LSTM(100, return_sequences=True,  activation='tanh'))
model.add(Dense(units=300, activation='relu'))
model.add(Dense(units=5, activation='linear'))
adam = Adam(lr=1e-4)
model.compile(loss='mse', optimizer=adam)

hello, I got problem connecting LSTM after Flatten layer.

Input 0 is incompatible with layer lstm_1: expected ndim=3, found ndim=2

It seems the lstm layer needs 3-d data and flatten layer only provides with 2-d data. So how can I modify it?

Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by Bigpig4396** [comment]
OK, I got it. I should use reshape layer instead of flatten layer.

**Comment by dunfrey** [comment]
> OK, I got it. I should use reshape layer instead of flatten layer.

Considering the question, if the input is an image with 128x128 dimensions. How to reshape to lstm?

**Comment by Bigpig4396** [comment]
you can use TimeDistributed
as you can check it from here
https://github.com/symoon94/DRQN-keras
